parlai/tasks/ubuntu/agents.py

```diff
@@ -46,3 +46,74 @@ def setup_data(self, path):
                     cands.append(response)
                     random.shuffle(cands)
                 yield (context, [response], None, cands), True
+
+
+class MultiTeacher(FixedDialogTeacher):
+    def __init__(self, opt, shared=None):
+        self.datatype = opt['datatype']
+        opt['datafile'] = os.path.join(
+            opt['datapath'], 'Ubuntu', opt['datatype'].split(':')[0] + '.csv'
+        )
+        super().__init__(opt, shared)
+
+        self.opt = opt
+        if shared:
+            self.data = shared['data']
+        else:
+            build(opt)
+            # '/Users/shaojie/src/ParlAI/data/Ubuntu/train.csv'
+            fold = opt.get('datatype', 'train').split(':')[0]
+            self._setup_data(fold)
+
+        self.num_exs = sum(len(d) for d in self.data)
```

Comment:
I believe this is going to have to be 2 * as well

Reply:
I think you're right. Then is that in DailyDialog task also needs to be `2 *`?
https://github.com/facebookresearch/ParlAI/blob/dfa982b5142b47a557196ea3524a5d6cb750396e/parlai/tasks/dailydialog/agents.py#L43

Reply:
Actually no, maybe not. Because in MOST teachers num_exs is all of the  (call, response) pairs.

In this teacher (and DailyDialogue), since we play both sides, we have a conversation like `a b c` and this produces (silence, a), (b, c) and (a, b). so it's the same as the length of the conversation. Other teachers already have things sectioned off into (call, response) pairs.

Apologies for directing you the wrong way...

Reply:
No worries, I'll changed it back ;)
What else should I do to merge? There is 1 failure reported below, but I don't know what to do about it.

Reply:
Right, just install black (`pip install black`) and run `black parlai/tasks/ubuntu/agents.py`. This will reformat the file and you can commit the changes.